internal/runner/runnerpool/runner.go

```diff
@@ -982,6 +987,33 @@ func applyPreventDestroyExclusions(l log.Logger, units []*component.Unit) {
 // maxDependencyTraversalDepth bounds the depth of dependency traversal to prevent excessive recursion.
 const maxDependencyTraversalDepth = 256
 
+// applyFilterAllowDestroyExclusions excludes units with destroy runs from Git-based filters
+// when the --filter-allow-destroy flag is not set. This prevents accidental destruction
+// of infrastructure when using Git-based filters.
+func applyFilterAllowDestroyExclusions(l log.Logger, opts *options.TerragruntOptions, units []*component.Unit) {
+	if opts.FilterAllowDestroy {
+		return
+	}
+
+	for _, unit := range units {
+		discoveryCtx := unit.DiscoveryContext()
+		if discoveryCtx == nil {
+			continue
+		}
+
+		hasRef := discoveryCtx.Ref != ""
+		hasDestroy := slices.Contains(discoveryCtx.Args, "-destroy")
```

Comment:
I think can be used isDestroyCommand()

Reply:
We can't because `isDestroyCommand` keys off opts.